util.go

```diff
@@ -191,6 +194,25 @@ func writeUDPConnMetadata(conn UDPConn, addr *net.UDPAddr, connMetadata *pb.Conn
 	return err
 }
 
+func sendPingMsg(conn UDPConn, num *uint64) {
+	pingMsg := new(pb.ConnectionMetadata)
+	pingMsg.IsPing = true
+	b, _ := proto.Marshal(pingMsg)
+
+	for {
+		err := writeUDPConnMetadata(conn, nil, pingMsg)
```

Comment:
Will this function call return error when conn is closed? We need to make sure this goroutine doesn't leak. Need to test this.